scripts/aws/monitor.py

```diff
@@ -0,0 +1,206 @@
+# --aws-profile (name of the profile to use for aws credentials)
+# --api-gw (name of the api_gateway)
+# --splunk-host (splunk host IP and port)
+# --splunk-token (splunk access token)
+# e.g.
+# python monitor.py --aws-profile test --api-gw api --splunk-host 10.2.1.2:99 --splunk-token xxx
+#
+
+import logging
+import os
+import time
+import argparse
+from zipfile import ZipFile
+import botocore.session
+import botocore.exceptions
+
+
+def create_api_alarm(cw_session, alarm_name, metric,
+                     namespace, stat, comparison, description,
+                     threshold, period, eval_period, dimensions, topic):
+    """Create the alarm for appropriate metric in API Gateway"""
+    response = cw_session.put_metric_alarm(
+        AlarmName=alarm_name,
+        AlarmDescription=description,
+        ActionsEnabled=True,
+        Namespace=namespace,
+        Dimensions=dimensions,
+        MetricName=metric,
+        Statistic=stat,
+        Period=period,
+        EvaluationPeriods=eval_period,
+        Threshold=threshold,
+        ComparisonOperator=comparison,
+        AlarmActions=[topic])
+
+    logging.info('response for creating api alarm = "%s"',
+                 response['ResponseMetadata'])
+
+
+def create_firehose_stream(firehose, stream_name, arn, bucket_arn):
+    """Setup a stream(if required) to export cloudwatch logs to s3"""
+    response = firehose.create_delivery_stream(
+        DeliveryStreamName=stream_name,
+        S3DestinationConfiguration={
+            'RoleARN': arn,
+            'BucketARN': bucket_arn,
+            'Prefix': '',
+            'BufferingHints': {
+                'SizeInMBs': 5,
+                'IntervalInSeconds': 300
+            },
+            'CompressionFormat': 'UNCOMPRESSED',
+            'EncryptionConfiguration': {
+                'NoEncryptionConfig': 'NoEncryption',
+            },
+            'CloudWatchLoggingOptions': {
+                'Enabled': False,
+            }
+        })
+
+    logging.info('response for creating firehose stream = "%s"',
+                 response)
+
+
+def create_lambda_function_zip(splunk_host, splunk_token):
+    """Updates and Zips the lambda function file"""
+
+    with open('index.js', "wt") as lambda_function_file:
+        with open('index.js.template', 'rt') as lambda_function_template:
+            for line in lambda_function_template:
+                lambda_function_file.write(line.replace('<splunk host:port>',
+                                                        splunk_host).replace('<token>',
+                                                                             splunk_token))
+    zip_file = 'lambda.zip'
+
+    with ZipFile(zip_file, 'w') as lambda_zip:
+        lambda_zip.write('index.js')
+
+    return zip_file
+
+
+def role_exists(iam, role_name):
+    """Checks if the role exists already"""
+    try:
+        iam.get_role(RoleName=role_name)
+    except botocore.exceptions.ClientError:
```

Comment:
Apparently not. It seems to be the only one.
https://github.com/boto/boto3/issues/167